### ISSUE TITLE
Quick enable/disable function

### ISSUE THREAD

**Issue by fpbark** [fphilipe/PageExtender.app#11]
Hi there! First of all, thx for making this. I've been looking for an alternative to Stylus for Safari and this does the job very well!

One of my main use cases with Stylus is to use custom CSS to filter search results on various web pages where the option is not available. For example, I'll hide certain page elements based on some text they contain. I find it very handy to have the ability to toggle Stylus on and off as I need it. 

Would it be possible to do this via some sort of toolbar icon?

**Comment by gingerbeardman** [comment]
@fpbark  love your Atari ST avatar!

**Comment by fpbark** [comment]
@gingerbeardman Thank you! When I was a kid, I really wanted a Mega Drive for Christmas (I want to say it was 1991) but my old man was adamant that I get a real computer instead— "you can play games AND learn how to use a computer." We ended up picking the 520STE over an Amiga 500. Very fond memories 😄

**Comment by fphilipe** [comment]
@fpbark are saying you want a toolbar icon to enable/disable the extension globally as if you were enabling/disabling it in the Safari Preferences?

**Comment by fpbark** [comment]
@fphilipe Yes. That would be ideal. A simple way to toggle the extension on and off from the toolbar. 

Since I’ve started using it, I’ve been keeping the Safari preferences window open at all times and enabling/disabling as needed.

**Comment by gingerbeardman** [comment]
I'm wondering why it's not enough to have different CSS/JS per site? That way, they're enabled per site automatically. That's the main point of this extension. 

Can you give some examples of your usage @fpbark ?

**Comment by fpbark** [comment]
Most of my use cases involve filtering searches on various sites. Here's one example: I collect vinyl records and spend a lot of time on Discogs. They don't offer a way to filter out sellers I've had bad experiences, certain versions of records that I already have, or artists that I'm not interested in. So I use both CSS and JS to create my own block lists that filter out page elements. At least a few times a day, I do want to see whether one my blocked sellers has an item I'm interested in, especially if it's a rare record. 

With Stylus in Chromium, this involves a click or two on the icon in my toolbar (the extension auto reloads the page when you enable/disable). With PageExtender, I have to open preferences, disable the extension, refresh the page, check if the item I want is there, enable the extension again and close preferences. I've been doing this multiple times a day since purchasing the extension from the App Store, or as I mentioned above, just leaving the preferences window open all the time.

I have the same use case across various websites albeit in slightly different capacities. It's great to be able to manipulate a page however you see fit but sometimes you still need to see the untouched original.

**Comment by fphilipe** [comment]
@fpbark, if the filtering happens purely with CSS, you could also inject some JS that renders a button or adds a keyboard shortcut that toggles a class on the root element. You can then include this class in your CSS filtering rules, such that the filtering only happens when the root class is currently set. That would allow you to toggle the filter without reloading the page.

You could even have such a JS in `default.js` such that the root class toggling is enabled on all pages.

A button that enables/disables the extension wouldn't be able to achieve this. As you've probably noticed, enabling the extension injects the JS/CSS into already open pages without needing to reload. Disabling the extension would require you to reload the page. If I'd implement such a toggle button, I'd not auto-reload the page as I consider this too intrusive (maybe you're filling out a form on that page, for instance).

**Comment by fpbark** [comment]
That sounds like a lot of work for what I think should be very basic functionality. Every other extension I use that changes the functionality or look/feel of a page (uBlock, Stylus, HTTPS Everywhere, Privacy Possum, Dark Reader etc), they all offer a simple way to quickly enable/disable their extensions (and they all re-load the page too FWIW). Seems like it's pretty standard behavior across the board. 

Anyway, no biggie. I'll fork the repo and play around with it myself to see if I can cobble something together.

**Comment by sn0wyfall** [comment]
@fphilipe Would it actually be possible to add in a button that just stops sites _globally_ from being themed? Or would that be too much work? In any case, thanks for the reply!

**Comment by fphilipe** [comment]
I'm open for adding such a button – eventually. I cannot share a timeline nor promise that this would ever see the light of day. Thanks for your suggestion and understanding 😊

**Comment by sn0wyfall** [comment]
No worries, thanks for the reply!

**Comment by fpbark** [comment]
@fphilipe Thx!
